diff --git a/libgringo/src/term.cc b/libgringo/src/term.cc
index 65c03c1..01ba0b9 100644
--- a/libgringo/src/term.cc
+++ b/libgringo/src/term.cc
@@ -21,6 +21,7 @@
 #include "gringo/term.hh"
 #include "gringo/logger.hh"
 #include "gringo/graph.hh"
+#include <cmath>
 
 namespace Gringo {
 
--- a/SConscript
+++ b/SConscript
@@ -221,8 +221,8 @@ if not conf.CheckMyFun('vsnprintf', 'char buf[256]; va_list args; vsnprintf (buf
     if conf.CheckMyFun('__builtin_vsnprintf', 'char buf[256]; va_list args; __builtin_vsnprintf (buf,256,"", args);', "#include <cstdio>\n#include <cstdarg>"):
         DEFS['vsnprintf']='__builtin_vsnprintf'
 
-if not conf.CheckMyFun('std::to_string', 'std::to_string(10);', "#include <string>"):
-    DEFS['MISSING_STD_TO_STRING']=1
+# if not conf.CheckMyFun('std::to_string', 'std::to_string(10);', "#include <string>"):
+#     DEFS['MISSING_STD_TO_STRING']=1
 
 env = conf.Finish()
 env.PrependUnique(LIBPATH=[Dir('.')])

