diff --git a/src/format/opamTypesBase.ml b/src/format/opamTypesBase.ml
index 8ec593f..1875d18 100644
--- a/src/format/opamTypesBase.ml
+++ b/src/format/opamTypesBase.ml
@@ -81,14 +81,7 @@ let parse_url (s,c) =
        | Some (proto1,proto2) ->
          (proto2^urlsplit^address, c), url_kind_of_string proto1
        | None ->
-         let addr = match proto with
-           | "git" -> s (* git:// urls legit *)
-           | _ ->
-             if Re_str.string_match (Re_str.regexp (".*"^urlsplit)) address 0
-             then address
-             else "http://" ^ address (* assume http transport by default *)
-         in
-         (addr,c), url_kind_of_string proto)
+         (s,c), url_kind_of_string proto)
     | [address] -> (address,c), `local
     | _ -> raise (Invalid_argument (Printf.sprintf "Bad url format %S" s))
 
