diff --git a/Makefile b/Makefile
index 8fba0d33..fe15a450 100644
--- a/Makefile
+++ b/Makefile
@@ -84,6 +84,7 @@ cstubs.subproject_deps = ctypes
 cstubs.deps = str bytes integers
 cstubs.install = yes
 cstubs.install_native_objects = yes
+cstubs.extra_hs = $(package_integers_path)/ocaml_integers.h
 
 cstubs: PROJECT=cstubs
 cstubs: $(cstubs.dir)/$(cstubs.extra_mls) $$(LIB_TARGETS)
diff --git a/Makefile.rules b/Makefile.rules
index c536845a..0dcdb733 100644
--- a/Makefile.rules
+++ b/Makefile.rules
@@ -80,8 +80,7 @@ INSTALL_CMIS = $($(PROJECT).public:%=$(BUILDDIR)/$($(PROJECT).dir)/%.cmi) \
 INSTALL_CMTIS = $($(PROJECT).public:%=$(BUILDDIR)/$($(PROJECT).dir)/%.cmti)
 INSTALL_CMTS = $($(PROJECT).public:%=$(BUILDDIR)/$($(PROJECT).dir)/%.cmt)
 INSTALL_MLIS = $($(PROJECT).public:%=$($(PROJECT).dir)/%.mli)
-INSTALL_HEADERS = $(wildcard $($(PROJECT).dir)/*.h) \
-                  $(package_integers_path)/ocaml_integers.h
+INSTALL_HEADERS = $(wildcard $($(PROJECT).dir)/*.h) $($(PROJECT).extra_hs)
 THREAD_FLAG = $(if $(filter yes,$($(PROJECT).threads)),-thread)
 LINK_FLAGS = $(as_needed_flags) $($(PROJECT).link_flags)
 OCAML_LINK_FLAGS=$(LINK_FLAGS:%=-cclib %)
